vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -545,3 +528,29 @@ func (e *StandardTxExecutor) BaseTx(tx *txs.BaseTx) error {
 	avax.Produce(e.State, e.Tx.ID(), tx.Outs)
 	return nil
 }
+
+// addStakerFromStakerTx creates the staker and adds it to state.
+func (e *StandardTxExecutor) addStakerFromStakerTx(stakerTx txs.Staker) error {
```

Comment:
refactor here since down the line, when stakers will be promoted to current immediately, this is the only code that will change